### ISSUE TITLE
Add option for `git add -p`

### ISSUE THREAD

**Issue by flimzy** [stevan/p5-App-Critique#19]
Some editors do automatic text formatting (especially as it relates to whitespace), or one may patch up some related code while fixing a perlcritic issue.

In such cases, the ability to run `git add -p` could be invaluable in separating the perlcritic-related fixes from other changes.

    Commit Message:
    
        Variable declared in conditional statement - critique(Variables::ProhibitConditionalDeclarations)
    
    Choose (y)es to use this message, or (n)o for more options [Y/n]: n
    ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
    What would you like to edit? (f)ile, (c)ommit message: 

Perhaps this prompt could add a `(p)atch` option to select which changes are included in the commit.

**Comment by stevan** [comment]
Yes, I was thinking about this as well because I have the same editor auto-cleanup issue as you. I had originally looked for a solution that just ignored whitespace changes, but that is not correct so I punted. I think the interactive patching could be a good compromise.
